route/route.go

```diff
@@ -32,67 +32,79 @@ func WithParam(ctx context.Context, p, v string) context.Context {
 	return context.WithValue(ctx, param(p), v)
 }
 
-// handle turns a Handle into httprouter.Handle
-func handle(h http.HandlerFunc) httprouter.Handle {
-	return func(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
-		ctx, cancel := context.WithCancel(context.Background())
+type contextFn func(r *http.Request) context.Context
+
+// Router wraps httprouter.Router and adds support for prefixed sub-routers
+// and per-request context injections.
+type Router struct {
+	rtr    *httprouter.Router
+	prefix string
+	ctxFn  contextFn
+}
+
+// New returns a new Router.
+func New(ctxFn contextFn) *Router {
```

Comment:
Do you envision a use case where a context function here would be useful instead of a base context?
Seems like values passed into the context would be static to a base context anyway. 

Seems like only a timeout for all requests applied to a new context would be sane.
Could be more explicit to configure as a directly. But both approaches seem equally fine.